frontend/packages/topology/src/components/quick-search/QuickSearch.tsx

```diff
@@ -12,11 +13,18 @@ interface QuickSearchProps {
 const QuickSearch: React.FC<QuickSearchProps> = ({ namespace, viewContainer }) => (
   <CatalogServiceProvider namespace={namespace}>
     {(catalogService: CatalogService) => (
-      <QuickSearchController
-        {...catalogService}
-        namespace={namespace}
-        viewContainer={viewContainer}
-      />
+      <QuickStartsLoader>
+        {(quickStarts, qsLoaded) => (
+          <QuickSearchController
+            catalogItems={catalogService.items}
+            quickStarts={quickStarts}
```

Comment:
All items should be generalized into a single data structure. In order to distinguish the origin of the results and abstract various providers in the search you can create a structure such as the following or come up with something else. The idea is to remove all the `if devcatalog else if quick start` conditions throughout the quick search and also support more providers in the future such as our samples or add page items.

```
[
  {
    catalog: 'devcatalog',
    getViewAllURL: (searchTerm: string) => `/catalog/ns/${namespace}?keyword=${searchTerm}`,
    // t('topology~View all developer catalog items ({{count, number}})')
    viewAllLinkKey: 'topology~View all developer catalog items ({{count, number}})'
  },
  {
    catalog: 'quickStarts',
    getViewAllURL: (searchTerm: string) => `/quickstart?keyword=${searchTerm}`,
    // t('topology~View all quick starts ({{totalQcItems, number}})'
    viewAllLinkKey: 'topology~View all quick starts ({{count, number}})',
  }
]
```

By generalizing each data set as a separate catalog, we can re-introduce the `catalog` property to the catalog extensions to distinguish providers for different catalogs (consult with @rohitkrai03).
Using this data structure you can create multiple `CatalogServiceProvider` wherein each provides items from their specific catalogs. This would require you to create a quick start catalog provider.

Whatever path we decide on, quick search should act on a generic data structure which provides all the necessary items, search capabilities, and linking for each item.

Reply:
Discussed with @rohitkrai03 and he has concerns with the extra work necessary to use the catalog for quick starts. Where the data comes from and how we use the data are separate enough that we can address the "where" later on and continue to use the `QuickStartsLoader`. However generalizing the data should still be addressed.

Reply:
Yeah, we can generalise the data structure for the quick search catalog with information about view all link and label. All this can be abstracted in the `QuickSearchController` so that we can easily move this logic to extensions and not affect other components.